components/tdisplays3/t_display_s3.cpp

```diff
@@ -8,26 +8,132 @@ namespace tdisplays3 {
 
 static const char *const TAG = "TDisplayS3";
 
+#ifdef TDISPLAYS3_USE_ASYNC_IO
+namespace writer_shim {
```

Comment:
Not wrong, but is the extra namespace really needed? And if it is, could you add the "  // end namespace xyz comment at its end? ;-)

Reply:
Yeah I though about that, but the writer-shim provides an interface between the class (the actual component), and the extra thread that it is running. The namespace is purely there to isolate the members of this namespace (clearification).

Another way would be to put this in a class, and pass the class reference as a parameter to the thread-start function.
I could rework towards this if you like. 
On a second thought, it would give the added possibility of having multiple of these displays running on the same ESP.

Reply:
I think you just answered yourself, that the class approach does have its advantages ;-)